torch/testing/_internal/common_distributed.py

```diff
@@ -515,14 +524,22 @@ def run_test(self, test_name: str, pipe) -> None:
         try:
             getattr(self, test_name)()
             # Close pipe after done with test.
-            pipe.close()
+            if signal_pipe is not None:
```

Comment:
In which cases would signal_pipe not be None? Shouldn't we update `distributed_test.py` to have this logic as well since it calls `run_test`?

Reply:
Well, it is _not_ None when we call it from the `_run` method of this very class, which I modified just a few lines above.

However it _is_ None when called from here:
https://github.com/pytorch/pytorch/blob/700df82881786f3560826f194aa9e04beeaa3fd8/torch/testing/_internal/distributed/distributed_test.py#L531

Note that TestDistBackend._run is currently _not_ starting any event listener thread, hence the signal pipe isn't relevant in that scenario. It might make sense to add the event listener to TestDistBackend too (or, at that point, to some common superclass) but that's beyond the fix I'm carrying out here.